[NewProduct] Contao Open Source CMS

bezin opened the pull request.

bezin committed:
Add Contao Open Source CMS

marcwrobel commented on products/contao:
```diff
@@ -0,0 +1,38 @@
+---
```
The file name is missing the `.md` extension. It should be `contao.md`.

marcwrobel replied:
Renamed.

marcwrobel commented on products/contao:
```diff
@@ -0,0 +1,38 @@
+---
+title: Contao
+category: server-app
+changelogTemplate: "https://github.com/contao/contao/blob/__RELEASE_CYCLE__/CHANGELOG.md"
```
`changelogTemplate` may be updated with the anchor to the specific version (similar to `docker-engine.md`).

marcwrobel replied:
Last version is always at the top of the document, no need to change the template to something more complex.

marcwrobel commented on products/contao:
```diff
@@ -0,0 +1,38 @@
+auto:
+  - git: https://github.com/contao/contao.git
+
+releases:
```
`releases` is missing versions 4.9 which is still supported (https://contao.org/en/release-plan.html).

Maybe all EOL LTS version should be documented too (https://en.wikipedia.org/wiki/Contao#Versions) ? @captn3m0 what do you think ?

bezin replied:
I added 4.9, too. I completely forgot about it, because the fact that there is two overlapping LTS versions is only now with 4.9 and 4.13 as the release schedule has been changed (LTS was extended). I am happy to add EOL LTS versions, too, if that is a requirement?

marcwrobel replied:
I do not know if it's a requirement to have EOL LTS version / EOL version between 4.9 and 4.13.

Let's start with supported version first and wait for @captn3m0's answer.

marcwrobel replied:
I added :

- versions between 4.9 and 4.13,
- EOL LTS versions having support / EOL dates on https://en.wikipedia.org/wiki/Contao#Versions.

I do not think adding more EOL versions is useful.

marcwrobel commented on products/contao:
```diff
@@ -0,0 +1,38 @@
+
+iconSlug: NA
+permalink: /contao
+releasePolicyLink: https://contao.org/en/release-plan.html
```
Seems like https://contao.org/release-plan.html is working and redirect to the user language (or at least the english language). https://contao.org/release-plan.html should be preferred over the 'localized' link.

marcwrobel replied:
Done.

marcwrobel commented on products/contao:
```diff
@@ -0,0 +1,38 @@
+releasePolicyLink: https://contao.org/en/release-plan.html
+releaseColumn: true
+releaseDateColumn: true
+versionCommand: composer show contao/core
```
It seems there are multiple installation method available (https://docs.contao.org/manual/en/installation/install-contao/). Does the command `composer show contao/core` work whatever the installation method used ?

If no, I think it is best to remove the `versionCommand`.

bezin replied:
You are right. The Contao Manager is basically a GUI for composer and it is for users who cannot or do not want to use the CLI. So the command would still work, but it is obviously not reachable for these people.

marcwrobel committed:
[Contao] Rename file

marcwrobel committed:
[Contao] Reorganize frontmatter

marcwrobel commented on products/contao.md:
```diff
@@ -0,0 +1,38 @@
+
+> [Contao Open Source CMS](https://contao.org) is a content management system written in PHP.
+
+Its LTS release is actively supported for three years + one year with security fixes. All other versions are supported for half a year.
```
There are multiple LTS releases at the same time. Maybe just `LTS releases are actively supported...`.

marcwrobel replied:
Done.

marcwrobel committed:
[Contao] Update description

marcwrobel committed:
[Contao] Use unlocalized changelogTemplate link

bezin committed:
Update contao.md

bezin committed:
Add activeSupportColumn

marcwrobel commented on products/contao.md:
```diff
@@ -0,0 +1,45 @@
+    latest: "5.0.7"
+    latestReleaseDate: 2022-11-30
+    lts: false
+    discontinued: false
```
`discontinued` properties must be removed. It is reserved mainly for devices, and `discontinuedColumn` is not visible so it is never used.

marcwrobel replied:
Done (@bezin sorry I had to override your commit to avoid conflicts).

marcwrobel committed:
[Contao] Remove discontinied properties

It is reserved mainly for devices, and discontinuedColumn is not visible.

marcwrobel committed:
[Contao] Add a few EOL versions

Added :

- versions between 4.9 and 4.13,
- EOL LTS versions having support / EOL dates on https://en.wikipedia.org/wiki/Contao#Versions.

marcwrobel reviewed (approved):
LGTM. @captn3m0 there is one final question for which I do not have the answer : should we add all the EOL versions or not ? See https://github.com/endoflife-date/endoflife.date/pull/1992#discussion_r1044289028 for more info.
